recipe_scrapers/__init__.py

```diff
@@ -605,22 +607,40 @@ def scraper_exists_for(url_path: str) -> bool:
     return host_name in get_supported_urls()
 
 
+def check_scraper_allowed(url_path, host_name, **options: Any) -> bool:
+    """Checks the website's robots.txt file to see if scraping is allowed.
+    If scraping is allowed, returns true. Note that this does not check to see
+    when the last time robots.txt was checked, so this could lead to constant pings
+    on robots.txt"""
+    if not options.get("robots.txt", True):
+        return True
+    else:
+        url = urljoin(host_name, 'robots.txt')
```

Comment:
I think we probably want to use `urljoin(url, '/robots.txt')` here - the `host_name` _should_ match the `url` hostname, but let's lean on the safe side.

```pyi
>>> from urllib.parse import urljoin
>>> urljoin('https://www.example.org/foo/bar', '/robots.txt')
'https://www.example.org/robots.txt'
```

(example to show that using the `/` prefix in `/robots.txt` results in the expected root-level path)